api/src/main/java/org/openmrs/module/queue/api/QueueServicesWrapper.java

```diff
@@ -104,14 +110,13 @@ public Concept getConcept(String conceptRef) {
 				return c;
 			}
 		}
-		//handle name
-		List<Concept> concepts = getConceptService().getConceptsByName(conceptRef);
-		if (concepts.size() == 1) {
-			return concepts.get(0);
-		} else if (concepts.size() > 1) {
-			throw new IllegalArgumentException("More than one concept is found with name: " + conceptRef);
+		//handle name.  this isn't ideal, as core will just log a warning if there are 2 concepts with the same name
+		//but there are no alternative suitable methods to get a List of Concepts that exactly match a given name
+		c = getConceptService().getConceptByName(conceptRef);
+		if (c == null) {
+			throw new IllegalArgumentException("Unable to find concept: " + conceptRef);
 		}
```

Comment:
I discovered in testing that the method I was previously using (`getConceptService().getConceptsByName(conceptRef)`)  does a fuzzy-match (eg. partial match) rather than an exact match, so it was matching on too many concepts.  Unfortunately, there is no public method in the ConceptService to get back all Concepts that exactly match a particular name, and the one above that I am using will just log a warning if multiple names are found and return the first one, which really isn't ideal, but given that it is also what is used in the new Core 2.6 method that accomplishes the same as the above, I went with it.  We likely should address this in core, and provide some additional (better) options.

Reply:
IIRC, the method in core was basically pulled straight from HTMLFormEntry and similarly didn't want to mess with what already worked. I agree having something like `getConceptsByExactName()` in core would actually be really, really useful for these cases where we're using concept names as concept identifiers.